src/store.rs

```diff
@@ -77,36 +49,61 @@ where
 }
 
 pub fn store_stream_for_range<PdC: PdClient>(
-    range: BoundRange,
+    range: (Vec<u8>, Vec<u8>),
+    pd_client: Arc<PdC>,
+) -> BoxStream<'static, Result<((Vec<u8>, Vec<u8>), Store)>> {
+    let bnd_range = BoundRange::from(range.clone());
+    pd_client
+        .stores_for_range(bnd_range)
+        .map_ok(move |store| {
+            let region_range = store.region.range();
+            let result_range = range_intersection(
+                region_range,
+                (range.0.clone().into(), range.1.clone().into()),
+            );
+            ((result_range.0.into(), result_range.1.into()), store)
+        })
+        .into_stream()
+        .boxed()
+}
+
+pub fn store_stream_for_range_by_start_key<PdC: PdClient>(
```

Comment:
it almost can, but making the return type correct is tricky. I'm pretty sure it can be done with some fancy generics and a couple of closures, but life is too short